apps/heft/src/cli/HeftCommandLineUtilities.ts

```diff
@@ -165,45 +160,53 @@ export class HeftCommandLineUtilities {
       actionParameterMap.set(action, parameter);
     }
 
-    const self: HeftCommandLineUtilities = this;
-    const parameterObject: IHeftBaseParameter<TValue> = {
-      get value(): TValue | undefined {
-        self._verifyParametersProcessed(options.parameterLongName);
-        if (self._commandLineParser.selectedAction) {
-          const parameter: TTSCommandLineParameter | undefined = actionParameterMap.get(
-            self._commandLineParser.selectedAction
-          );
-          if (parameter) {
-            return getParameterValue(parameter);
+    const parameterObject: IHeftBaseParameter<TValue> = Object.defineProperties(
+      {} as IHeftBaseParameter<TValue>,
+      {
+        value: {
+          get: (): TValue | undefined => {
+            this._verifyParametersProcessed(options.parameterLongName);
+            if (this._commandLineParser.selectedAction) {
+              const parameter: TTSCommandLineParameter | undefined = actionParameterMap.get(
+                this._commandLineParser.selectedAction
+              );
+              if (parameter) {
+                return getParameterValue(parameter);
+              }
+            }
+
+            return undefined;
           }
-        }
+        },
 
-        return undefined;
-      },
+        actionAssociated: {
+          get: (): boolean => {
+            if (this._commandLineParser.selectedAction) {
+              if (actionParameterMap.get(this._commandLineParser.selectedAction)) {
+                return true;
+              }
+            }
 
-      get actionAssociated(): boolean {
-        if (self._commandLineParser.selectedAction) {
-          if (actionParameterMap.get(self._commandLineParser.selectedAction)) {
-            return true;
+            return false;
```

Comment:
Can be simplified to
`return this._commandLineParser.selectedAction && actionParameterMap.get(this._comandLineParser.selectedAction)`